src/Generators/Microsoft.Gen.AutoClient/Parser.cs

```diff
@@ -653,16 +654,19 @@ private void Diag(DiagnosticDescriptor desc, Location? location, params object?[
         _reportDiagnostic(Diagnostic.Create(desc, location, messageArgs));
     }
 
+    [ExcludeFromCodeCoverage]
```

Comment:
Is it because the compiler generates some code that is never accessed?

If so I would assume that compiler generated code could be excluded and the tool should have an option for that.

Reply:
I filed a bug about this against Roslyn. Not all the generated code in a record type is properly annotated. The properties for example don't have annotations. So it messes up coverage.